connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -141,17 +141,35 @@ public void initialize(TaskConfig taskConfig) {
     public void stop() {
         // Offset commit is handled upon exit in work thread
         super.stop();
+        try {
+            task.stop();
+        } catch (Exception e) {
+            log.warn("Could not stop task", e);
+        }
         consumer.wakeup();
     }
 
     @Override
     protected void close() {
         // FIXME Kafka needs to add a timeout parameter here for us to properly obey the timeout
         // passed in
-        task.stop();
-        if (consumer != null)
-            consumer.close();
-        transformationChain.close();
+        try {
+            task.stop();
+        } catch (Exception e) {
+            log.warn("Could not stop task", e);
+        }
+        if (consumer != null) {
+            try {
+                consumer.close();
+            } catch (Exception e) {
+                log.warn("Could not stop consumer", e);
+            }
+        }
+        try {
+            transformationChain.close();
+        } catch (Exception e) {
+            log.warn("Could not stop transformation chain", e);
+        }
```

Comment:
Should these all be in finally blocks? It gets messy, but it seems like closing resources is mandatory even when there are errors. Or, catch `Throwable` rather than `Exception`?

Reply:
I'll change to `Throwable`.  I did a grep of `catch (Throwable)` in the Kafka repo and it looks like it is more prevalent in Connect.  Makes sense since Connect deals with external systems which are out of our control.